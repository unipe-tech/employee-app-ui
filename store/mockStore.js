const store = {
  // Auth

  auth: {
    id: 1,
    phoneNumber: "9876543210",
    verifyStatus: "PENDING",
  },

  // Aadhaar

  aadhaar: {
    data: {
      aadhaar_data: {
        name: "Xyz garg",
      },
    },
    number: "789456123010",
    submitOTPtxnId: "987",
    verifyMsg: "This otp has been sent",
    verifyStatus: "PENDING",
  },

  // Bank

  bank: {
    accountNumber: "157894561230",
    ifsc: "INDB0000123",
    upi: "9876543210@upi",
    accountHolderName: "Xyz Goyal",
    verifyStatus: "PENDING",
    verifyMsg: "The verification is pending",
    bankName: "IndusInd Bank",
    bankBranch: "Mumbai",
    branchCity: "Mumbai",
  },

  // ESIC

  esic: {
    portal: { estCode: "", ipNumber: "" },
    familyDetails: {
      fatherHusband: { relation: "", name: "" },
      nominee: { relation: "", name: "" },
    },
    address: {
      present: { street: "", state: "", district: "", pincode: "" },
      permanent: { street: "", state: "", district: "", pincode: "" },
      nominee: { street: "", state: "", district: "", pincode: "" },
    },
  },

  // License

  license: {
    number: "",
    name: "",
    dob: "",
    bloodGroup: "",
    classes: {},
    rto: {},
    validity: {},
    verifyMsg: "",
    verifyStatus: "PENDING",
  },

  // Navigation

  navigation: { currentScreen: "Welcome" },

  // PAN

  pan: {
    dob: "02/02/2002",
    email: "xyz@unipe.co",
    gender: "Male",
    name: "Xyz Goyal",
    number: "ABCDE1234F",
    verifyMsg: "Verification Pending",
    verifyStatus: "PENDING",
  },

  // Profile

  profile: {
    maritalStatus: "Married",
    educationalQualification: "12th pass",
    alternatePhone: "+913216549870",
    email: "abc@unipe.co",
    selfie: "",
  },

  // Timer

  timer: { login: 2 * 60, aadhaar: 10 * 60 },
};

export default store;
