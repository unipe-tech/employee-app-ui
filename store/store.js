import { configureStore } from "@reduxjs/toolkit";
import authSlice from "./slices/authSlice";
import aadhaarSlice from "./slices/aadhaarSlice";
import panSlice from "./slices/panSlice";
import bankSlice from "./slices/bankSlice";
import esicSlice from "./slices/esicSlice";
import profileSlice from "./slices/profileSlice";
import navigationSlice from "./slices/navigationSlice";

export const store = configureStore({
  reducer: {
    auth: authSlice,
    aadhaar: aadhaarSlice,
    pan: panSlice,
    bank: bankSlice,
    profile: profileSlice,
    esic: esicSlice,
    navigation: navigationSlice
  },
});
