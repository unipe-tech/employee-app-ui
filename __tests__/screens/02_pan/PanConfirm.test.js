import React from "react";
import { render, fireEvent, waitFor, act } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { Provider } from "react-redux";
import { store } from "../../../store/store";
import { shallow } from "enzyme";
import { Button } from "@react-native-material/core";
import { Alert } from "react-native";
import spyOnAlert from "../../../testHelpers/spyOnAlert";
import PanConfirm from "../../../screens/02_pan/PanConfirm";

const { pressAlertButton } = spyOnAlert();

describe("PanConfirm Screen", () => {
  describe("Back Button Testing", () => {
    it("testing UI rendering", () => {
      const tree = shallow(
        <Provider store={store}>
          <NavigationContainer>
            <PanConfirm />
          </NavigationContainer>
        </Provider>
      );
      expect(tree).toMatchSnapshot();
    }, 8000);
  });
});
