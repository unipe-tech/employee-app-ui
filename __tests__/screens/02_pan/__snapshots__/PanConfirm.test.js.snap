// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanConfirm Screen Back Button Testing testing UI rendering 1`] = `
<ContextProvider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ForwardRef(NavigationContainerInner)>
    <PanConfirm />
  </ForwardRef(NavigationContainerInner)>
</ContextProvider>
`;
